emulator.c

```diff
@@ -64,8 +98,19 @@ execute_instruction(i8080 *cpu, uint8_t opcode)
       printf("MVI H");
       break;
     case 0x29: // NOLINT
-      printf("DAD H");
-      break;
+      {        // DAD H
+        uint32_t sum = cpu->l;
+        sum = (sum << 8) + cpu->h; // NOLINT
+        sum = sum << 1;            // double hl
```

Comment:
I think this is backwards as well? H is the upper register so should be sum = (cpu->h << 8) + cpu->l to get the HL value?

Reply:
I just did this op with a different register, and I interpreted it this way with H as the upper reg (cpu->h << 8) for upper bits.  Let me know if it's actually the other way around!